packages/components/src/components/hds/popover-primitive/index.ts

```diff
@@ -57,36 +99,42 @@ export default class HdsPopoverPrimitiveComponent extends Component {
     }
   }
 
-  setupPrimitiveContainer = modifier(
-    (element) => {
+  setupPrimitiveContainer = modifier<SetupPrimitiveContainerModifier>(
+    (element: HTMLElement) => {
       this.containerElement = element;
 
       // we register the "soft" events
       if (this.enableSoftEvents) {
+        // @ts-expect-error: known issue with type of invocation
         registerEvent(this.containerElement, ['mouseenter', this.onMouseEnter]);
```

Comment:
Last time I checked out this PR these were the only errors returned by TypeScript linter. I take it we couldn't find a way around it, right?

Reply:
Yes, this is something I wanted to ask @aklkv too. With the patch you created, this error went away, but removing the patch this is still something TS complains about. Can you expand on why this is?

Reply:
The patch only tried addressing missing options for `eager` and as we learned it's no longer needed and we don't need this patch or need the `eager` option. When it comes to the comment above I have asked the question on ember Discord https://discord.com/channels/480462759797063690/1256103260943552512/1256103588430614559
so this is legal use case but not totally covered by community nor it has piping in place. So I would keep an eye on this and it might get better TS support as I mentioned we might need something like `invokeHelper`